README.md

```diff
@@ -9,39 +9,43 @@ At midday, no matter if it is a weekend or a holiday, a tip will be posted and t
 > 12/29/2015
 
-Insert an item into an existing array is a daily common task. You can add elements to the end of an array using push, to the beginning using unshift, or the middle using splice.
-But those known methods doesn't mean that are the more porformant, here we go...
-
-Add a element at the end of the array is easy with push(), but there are a way more performant.
+Inserting a new item into an existing array is a common daily task. You may append elements to the array using [`Array.prototype.push()`](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/push), prepend them using [`Array.prototype.unshift()`](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift), or insert them inbetween using [`Array.prototype.splice()`](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/splice).
+But although these are the best known methods for those tasks, better performing ways do exist.
 
+Adding an element at the end of the array is easy with [`Array.prototype.push()`](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/push) but there's a faster alternative:
 ```javascript
-var arr = [1,2,3,4,5];
+var items = [1,2,3,4,5];
 
-arr.push(6);
-arr[arr.length] = 6; // 43% faster in Chrome 47.0.2526.106 on Mac OS X 10.11.1
+items.push(6);
+items[items.length] = 6; // 43% faster in Chrome 47.0.2526.106 on Mac OS X 10.11.1
 ```
-Both methods modify the original array. Doesn't believe me? Check the [jsperf](http://jsperf.com/push-item-inside-an-array)
-
-Now we are trying to add a item to the beginning of the array 
+Don't believe me? Check the [jsperf](http://jsperf.com/push-item-inside-an-array)!
+Both methods will modify the original array. 
 
+Lets prepend an item to the the array:
 ```javascript
-var arr = [1,2,3,4,5];
+var items = [1,2,3,4,5];
 
-arr.unshift(0);
-[0].concat(arr); // 98% faster in Chrome 47.0.2526.106 on Mac OS X 10.11.1
+items.unshift(0);
+[0].concat(items); // 98% faster in Chrome 47.0.2526.106 on Mac OS X 10.11.1
 ```
-Here a little bit detail, unshift edit the original array, concat return a new array. [jsperf](http://jsperf.com/unshift-item-inside-an-array)
-
-Add items at the middle of an array is easy with splice and is the most performant way to do it.
+Instead of using unshift we can also create a fresh array and join it with `arr`. [`Array.prototype.concat()`](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/push) will combine both into a new array and return the result. 
+Check the [jsperf](http://jsperf.com/unshift-item-inside-an-array)!
 
+For inserting elements inbetween the existing ones [`Array.prototype.splice()`](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) turns out to be the best option.
 ```javascript
 var items = ['one', 'two', 'three', 'four'];
 items.splice(2, 0, 'hello');
```

Comment:
update this line `items.splice(2, 0, 'hello');` by `items.splice(items.length, 0, 'hello');`